05-JWT-Basics/preferred/public/browser-app.js

```diff
@@ -0,0 +1,70 @@
+const formDOM = document.querySelector('.form')
+const usernameInputDOM = document.querySelector('.username-input')
+const passwordInputDOM = document.querySelector('.password-input')
+const formAlertDOM = document.querySelector('.form-alert')
+const resultDOM = document.querySelector('.result')
+const btnDOM = document.querySelector('#data')
+const tokenDOM = document.querySelector('.token')
+
+formDOM.addEventListener('submit', async (e) => {
+  formAlertDOM.classList.remove('text-success')
+  tokenDOM.classList.remove('text-success')
+
+  e.preventDefault()
+  const username = usernameInputDOM.value
+  const password = passwordInputDOM.value
+
+  try {
+    const { data } = await axios.post('/api/v1/login', { username, password })
+
+    formAlertDOM.style.display = 'block'
+    formAlertDOM.textContent = data.message
+
+    formAlertDOM.classList.add('text-success')
+    usernameInputDOM.value = ''
+    passwordInputDOM.value = ''
+
+    localStorage.setItem('token', data.token)
+    resultDOM.innerHTML = ''
+    tokenDOM.textContent = 'token present'
+    tokenDOM.classList.add('text-success')
+  } catch (error) {
+    formAlertDOM.style.display = 'block'
+    formAlertDOM.textContent = error.response.data.message
+    localStorage.removeItem('token')
+    resultDOM.innerHTML = ''
+    tokenDOM.textContent = 'no token present'
+    tokenDOM.classList.remove('text-success')
+  }
+  setTimeout(() => {
+    formAlertDOM.style.display = 'none'
+  }, 2000)
+})
+
+btnDOM.addEventListener('click', async () => {
+  const token = localStorage.getItem('token')
+  try {
+    const { data } = await axios.get('/api/v1/dashboard', {
```

Comment:
This route doesn't exist in the preferred directory.

Reply:
Fixed it